### ISSUE TITLE
Unable to integrate with Storybook ({ apply: [Function: bound apply] })

### ISSUE THREAD

**Issue by astrotim** [webpack-contrib/stylelint-webpack-plugin#162]
* Operating System: OSX 10.11.6 El Capitan
* Node Version: 8.10.0
* NPM Version: 5.6.0
* webpack Version: 3.12.0
* stylelint-webpack-plugin Version: 0.10.5

This issue is for a:

- [ ] **bug**
- [ ] **feature** request
- [ ] **modification** request

### Code

Note: Storybook exposes a method for modifying the Webpack configuration [documented here](https://storybook.js.org/configurations/custom-webpack-config/#full-control-mode)

##### webpack.config.js (within .storybook/ directory)

```js
const StyleLintPlugin = require('stylelint-webpack-plugin');

module.exports = storybookBaseConfig => {
  storybookBaseConfig.plugins.push(new StyleLintPlugin({
      extends: 'stylelint-config-standard'
    })
  );
  console.log(storybookBaseConfig.plugins);
  return storybookBaseConfig;
};
```

### Expected Behavior

I expect Stylelint to run against my CSS files. 

I expect to see `StyleLintPlugin` as part of the `plugins` array, as per the console.log in the code snippet above.

### Actual Behavior

Stylelint does not run.

The plugins array logged in my console lists `{ apply: [Function: bound apply] }`

```
[...
  ProgressPlugin { profile: undefined, handler: undefined },
  DefinePlugin { definitions: {} },
  { apply: [Function: bound apply] }  // ← ???
]
```

### Testing other Webpack plugins

If I push a different plugin, for example the EnvironmentPlugin, I see this logged in my console as expected.

```js
const StyleLintPlugin = require('stylelint-webpack-plugin');
const webpack = require('webpack');

module.exports = storybookBaseConfig => {
  storybookBaseConfig.plugins.push(
    new StyleLintPlugin({
      extends: 'stylelint-config-standard'
    })
  );
  storybookBaseConfig.plugins.push(
    new webpack.EnvironmentPlugin(['NODE_ENV', 'DEBUG'])
  );
  console.log(storybookBaseConfig.plugins);
  return storybookBaseConfig;
};
```

```
[...
  ProgressPlugin { profile: undefined, handler: undefined },
  DefinePlugin { definitions: {} },
  { apply: [Function: bound apply] }, // ← ???
  EnvironmentPlugin { keys: [ 'NODE_ENV', 'DEBUG' ], defaultValues: {} } // ← expected
]
```

### How Do We Reproduce?

Create a Storybook application and implement a custom Webpack configuration as per [these docs](https://storybook.js.org/configurations/custom-webpack-config) and implement the code above.

**Comment by shellscape** [comment]
This seems like a request for support for integrating this with a third party. Citing the first paragraph of the issue template:

> 👉🏽 Need support, advice, or help? Don't open an issue!
  Head to StackOverflow or https://gitter.im/webpack/webpack.

Since the documentation and all mentions of the issue's context originate from/in Storybook, I'd suggest debugging with the team or userbase for that project first, and then returning here and pinging us if you've been able to identify an actual bug in this project's code.
